### PR TITLE
[DEV-1744] Remove `dtype` from feature creation payload

### PR DESCRIPTION
**Pull request by gee-senbong**
## Description

This PR aims to remove `dtype` from feature creation payload.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

**Review by yanhuisg — APPROVED** [review]
LGTM

**Review by kchua78 — APPROVED** [review]

File: featurebyte/models/feature.py [hunk 1]
```diff
@@ -192,22 +192,14 @@ class FeatureModel(FeatureByteCatalogBaseDocumentModel):
         When the Feature get updated
     """
 
-    dtype: DBVarType = Field(
-        allow_mutation=False, description="database variable type for the feature"
-    )
+    dtype: DBVarType = Field(allow_mutation=False, default=DBVarType.UNKNOWN)
```
Comment (YS-L) [inline featurebyte/models/feature.py:195]: Curious in actual code path, will the dtype ever be not provided when creating a FeatureModel?

**Review by YS-L — COMMENTED** [review]
Changes looks good 👍 

One question is that current when creating Feature objects through the various groupby like methods, the dtype is still passed in: https://github.com/featurebyte/featurebyte/blob/ff3950fd1a1d4d21a61d1dfa06bde45dc01e5797/featurebyte/api/view.py#L1353-L1360

Is that still required?

**Comment by gee-senbong** [comment]
> Changes looks good 👍
> 
> One question is that current when creating Feature objects through the various groupby like methods, the dtype is still passed in:
> 
> https://github.com/featurebyte/featurebyte/blob/ff3950fd1a1d4d21a61d1dfa06bde45dc01e5797/featurebyte/api/view.py#L1353-L1360
> 
> Is that still required?

@YS-L `Feature` in api object is different from `FeatureModel` in `models` directory. Since `dtype` is required for the `Series` object, we need to pass in the value there. However, during feature creation, there is no need to pass `dtype` as it will be derived at the feature model level.

**Pull request merged**